src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -492,6 +500,358 @@ TEST(Aktualizr, FullWithUpdatesNeedReboot) {
+  auto storage = INvStorage::newStorage(conf.storage);
+
+  std::vector<std::string> expected_event_order = {
+      "UpdateCheckComplete",    "DownloadProgressReport", "DownloadTargetComplete", "DownloadProgressReport",
+      "DownloadTargetComplete", "AllDownloadsComplete",   "InstallStarted",         "InstallTargetComplete",
+      "InstallStarted",         "InstallTargetComplete",  "AllInstallsComplete",    "PutManifestComplete"};
+
+  std::vector<std::string> expected_report_order = {
+      "EcuDownloadStarted",     "EcuDownloadCompleted",     "EcuDownloadStarted",     "EcuDownloadCompleted",
+      "EcuInstallationStarted", "EcuInstallationCompleted", "EcuInstallationStarted", "EcuInstallationCompleted"};
+
+  const std::string primary_ecu_id = "CA:FE:A6:D2:84:9D";
+  const std::string secondary_ecu_id = "secondary_ecu_serial";
+
+  {
+    fiu_init(0);
+    fiu_enable("fake_package_install", 1, nullptr, 0);
+
+    Aktualizr aktualizr(conf, storage, http_server_mock);
+    EventHandler event_hdlr(aktualizr);
+    aktualizr.Initialize();
+
+    // verify currently installed version
+    std::vector<Uptane::Target> installed_versions;
+    size_t current_version{SIZE_MAX};
+    size_t pending_version{SIZE_MAX};
+
+    ASSERT_TRUE(
+        storage->loadInstalledVersions(primary_ecu_id, &installed_versions, &current_version, &pending_version));
+
+    ASSERT_TRUE(installed_versions.empty());
+    ASSERT_EQ(pending_version, SIZE_MAX);
+    ASSERT_EQ(current_version, SIZE_MAX);
+
+    aktualizr.UptaneCycle();
+    aktualizr.uptane_client_->completeInstall();
+
+    ASSERT_TRUE(event_hdlr.checkRecievedEvents(expected_event_order));
+    ASSERT_FALSE(aktualizr.uptane_client_->hasPendingUpdates());
+    ASSERT_TRUE(http_server_mock->checkRecievedReports(expected_report_order));
+    ASSERT_FALSE(http_server_mock->wasInstallSuccessful(primary_ecu_id));
+    ASSERT_TRUE(http_server_mock->wasInstallSuccessful(secondary_ecu_id));
+
+    data::InstallationResult dev_installation_res;
+    std::string report;
+    std::string correlation_id;
+
+    // `device_installation_result` and `ecu_installation_results` are cleared
+    // at UptaneCycle()->putManifest() once a device manifest is successfully sent to a server
+    ASSERT_FALSE(storage->loadDeviceInstallationResult(&dev_installation_res, &report, &correlation_id));
+
+    std::vector<std::pair<Uptane::EcuSerial, data::InstallationResult>> ecu_installation_res;
+    EXPECT_FALSE(storage->loadEcuInstallationResults(&ecu_installation_res));
+    EXPECT_EQ(ecu_installation_res.size(), 0);
+
+    ASSERT_TRUE(
+        storage->loadInstalledVersions(primary_ecu_id, &installed_versions, &current_version, &pending_version));
+    // it says that no any installed version found,
+    // which is, on one hand is correct since installation of the found update failed hence nothing was installed,
+    // on the other hand some version should have been installed prior to the failed update
```

Comment:
True. Thanks for leaving this comment. If we change this functionality in the future, this should help make it more obvious what to do with this test.